### DISCUSSION TITLE
Docker Compose prerequisites for the Postal installation (MariaDB, RabitMQ)

### DISCUSSION THREAD

**Discussion by zdeneksvarc** [postalserver/postal#2147] [Help with using Postal]
Maybe it'll help someone
- https://github.com/zdeneksvarc/postal-storage

**Comment by willpower232** [comment]
Your repository is empty, did you forget to push?

**Reply by zdeneksvarc** [reply]
I just recreated the repository, you hit 20 seconds of unavailability 😄

**Reply by willpower232** [reply]
aha so I did, good to have a quick reference for those things 🙏

**Comment by Ascar699** [comment]
@zdeneksvarc What is the benefit of using postal-storage vs pulling the images as per the official documentation?
One observation I made while installing Postal multiple times - RabbitMQ is quite picky about passwords' length and characters used in it.
I was scratching my head when I was getting exit1 error on postal_web_1 until I changed the password from 20 characters to 10.

**Reply by zdeneksvarc** [reply]
The advantage over the official manual is the use of the [Docker Compose](https://docs.docker.com/compose/), which is more clear and systematic. Especially for larger deployments.

For RabitMQ, I normally use 16 character passwords.

**Comment by bonest** [comment]
Hi @zdeneksvarc thanks for making a composer file for mariadb and RabitMQ. 

But please tell me what you mean by this 'and set working directory cd /opt/postal-tls' 

What do you mean by set the working directory to cd /opt/postal-tls ?

**Reply by zdeneksvarc** [reply]
Edit, TLDR: `cd /opt/postal-storage` is the right path

Hi @bonest , it doesn't mean anything special, just make a ~~`cd /opt/postal-tls`~~ `cd /opt/postal-storage`. Because now you have an `.env` file that needs to be modified (MYSQL_ROOT_PASSWORD, RABBITMQ_DEFAULT_PASS).

**Reply by bonest** [reply]
Thanks but I still don't get it :-) Do you mean that I should copy the cloned repository to opt/postal-tls and alter the .env file there? out of the cloned repository?

I am in the process of installing postal, so I don't have a postal-tls folder yet, because the mariadb should be running before installing postal, and I have run into problems by running the container with the example in the install guide. It looks like its cutting off my password in the terminal, and when I look at the docker ps the mariadb continues to restart. 

So I thought why not use docker-compose, since its a much cleaner way of creating the configuration, than simply just running the container with all settings passed on the command line. 

Then I found this thread, and was happy to see a composer file with ports and external storage already defined, and cloned the repository but do not understand the cd opt/postal-tls as working direcotory

So please ELI5 (explain like I am 5) 

Thanks for being patient with me :-)

**Reply by zdeneksvarc** [reply]
Yes, you're right. ~~It can be a little confusing~~. I'll revise the text in about two hours. Thank you for the suggestion.

**Reply by zdeneksvarc** [reply]
I have corrected the [README.md](https://github.com/zdeneksvarc/postal-storage) file so that it now contains the correct information. It was the `postal-storage` folder, `not postal-tls`.

You can only have an love affair with the [postal-tls](https://github.com/zdeneksvarc/postal-tls) folder after installing Postal 🙃

Thanks again for the suggestion 🙏

**Comment by bonest** [comment]
Hi @zdeneksvarc Are you sure the volume mappings in the docker-compose file is correct?

I used the docker-compose and checked the /var/lib directories, and they did not exists after launching the composer file?
I had expected to see some files in the local mounts if they were mounted correctly?

Cheers.

**Reply by zdeneksvarc** [reply]
It depends on what Docker image you are using. In our `docker-compose.yml`, the images used are: 

- mariadb:10.9
- rabbitmq:3.8

In which the specified path **INSIDE** the container is valid.

**Reply by bonest** [reply]
My bad.. I have to learn a little more docker I guess :-). 

The data is persisted as it should be inside docker/volumes. (I was looking at my local /var/lib not the docker volumes) :-)
